wasmtime-api/src/instance.rs

```diff
@@ -108,6 +108,12 @@ impl Instance {
         let mut mutable = instance_handle.clone();
         for (name, _) in instance_handle.clone().exports() {
             let export = mutable.lookup(name).expect("export");
+            if let wasmtime_runtime::Export::Function { signature, .. } = &export {
+                // HACK ensure all handles, instantiated outside Store, present in
+                // the store's SignatureRegistry.
```

Comment:
Is it possible and valid to have a handle instantiated outside Store? I would expect that all instances that can interface  with each other must be in the same Store.

Reply:
All except the wasi InstanceHandles -- we import them using this method (`from_handle`). I'll add this in the comment.